packages/core/integration-tests/test/scope-hoisting.js

```diff
@@ -645,7 +658,7 @@ describe('scope hoisting', function() {
       assert.strictEqual(output, 2);
     });
 
-    it('throws a meaningful error on undefined exports', async function() {
+    it.skip('throws a meaningful error on undefined exports', async function() {
```

Comment:
So
```
export {Test};
```
is not a parse error with swc? Should we try to change that? It's a `SyntaxError`

Reply:
Seems to depend on the parser. It's a syntax error in Babel but not TSC or esprima, and also not an error in SWC. I have a feeling that adding this scope tracking to the parser would be a big perf hit.

> I guess we could detect this in esm2cjs or the hoist transform

Sure seems like a nice thing to do, but I'd prefer it to not block this PR.